docassemble/ALToolbox/data/questions/al_income.yml

```diff
@@ -117,6 +117,26 @@ subquestion: |
 
   ${ x.add_action() }
 ---
+generic object: ALItemizedJob
+continue button field: x.to_add.revisit
+question: |
+  Edit incomes
+subquestion: |
+  ${ x.to_add.table }
```

Comment:
Seeing a few things here I haven't caught before or that I'm perhaps misunderstanding. Sorry either way.

There are no review screens in the demo from what I can tell, so people don't get to see the review screens for the items. I didn't quite have time to write one myself, sorry, so I didn't get to see the new tables. I can try to do that another time. My comments here are based just on reading code.

1. Doesn't break this PR, but is tangentially relevant to it: I'm not sure there's any `x.table` for the type of object that this is, which is an `ALItemizedValueDict`.
2. Future issue: I see these edit screens for ALItemizedJobList as well, but I don't think that class has those properties, just the jobs inside have those properties.
3. Future issue: I think I'm actually seeing some missing tables in the code in general and at least one repeated item - the table for `ALExpenseList`, which I see at lines 614 and 161.

Reply:
> Couldn't even trigger it with itemized_job.to_add.revisit

ALItemizedJob doesn't have a revisit screen itself. It splits up better as a row in the ALItemizedJobList revisit screen; then if you edit the job, you can change the job title, employer info, and then get two separate screens for editing incomes from the ValueDict and editing deductions from the ValueDict. TBH most people won't be using ALItemizedJob by itself: they should be using the list, which does have a revisit screen.

> I'm not sure there's any x.table for the type of object that this is, which is an ALItemizedValueDict

There is, right above this, lines 111-118.

> I see these edit screens for ALItemizedJobList as well, but I don't think that class has those attributes, just the jobs inside have those attributes

I assume you're talking about the `ALItemizedJobList.table` (lines 468- 485)? That's how tables work: everything in the edit list are attributes of the things in the list.

> I think I'm actually seeing some missing tables in the code in general, and at least one repeated item - the table for ALExpenseList, which I see at lines 614 and 161.

Thanks! The repeated item is addressed now, and I added tables for ALIncomeList and ALAssetList. I don't have any examples of ALVehicleList or ALSimpleValueList yet, so I won't add those until I have a solid understanding of them (likely when I add those docs).

Reply:
> lines 468- 485

If those are the relevant tables, I think line 480 might need adjustment. It uses `x.table` instead of `x.to_add.table`:
```
generic object: ALItemizedJobList
continue button field: x.to_add.revisit
question: |
  Edit incomes from ${ x.source }
subquestion: |
  ${ x.table }
  ${ x.add_action() }
```
From what I can see, this is different than the `x.to_subtract.revisit` block.

Reply:
Another good catch. These were definitely bugged and not appearing, which is why I added these blocks (120 - 139). I'm going to delete them, and maybe reorganize things a bit.